features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/ServiceTracker.java

```diff
@@ -0,0 +1,483 @@
+            }
+        }
+    }
+
+    @EventHandler(ueis = {EventConstants.NODE_GAINED_SERVICE_EVENT_UEI,
+                          EventConstants.RESUME_POLLING_SERVICE_EVENT_UEI})
+    public void nodeGainedServiceHandler(final IEvent event) {
+        final Service service = Service.fromEvent(event);
+
+        final Optional<E> element = this.filterService.apply(service);
+        if (!element.isPresent()) { // TODO fooker: this smells
+            return;
+        }
+
+        if (!this.network.add(service, element.get())) {
+            LOG.debug("Service already known: {}", service);
+            return;
+        }
+
+        this.addService.accept(new ServiceEntry<>(service, element.get())); // TODO fooker: this smells
+    }
+
+    @EventHandler(uei = EventConstants.INTERFACE_REPARENTED_EVENT_UEI)
+    public void interfaceReparentedHandler(final IEvent event) {
+        final Node oldNode = new Node(Integer.parseInt(event.getParm(EventConstants.PARM_OLD_NODEID).getValue().getContent()));
+        final Node newNode = new Node(Integer.parseInt(event.getParm(EventConstants.PARM_NEW_NODEID).getValue().getContent()));
+
+        final InetAddress ipAddress = InetAddressUtils.addr(event.getInterface());
+
+        final Interface oldInterface = oldNode.iface(ipAddress);
+        final Interface newInterface = newNode.iface(ipAddress);
+
+        // Remove interface from old node and add to new one
+        this.network.remove(oldInterface).forEach(this.delService.andThen(oldEntry -> {
+            final Service newService = newInterface.service(oldEntry.service.serviceName);
+
+            final Optional<E> newElement = this.filterService.apply(newService);
+            if (!newElement.isPresent()) { // TODO fooker: this smells
+                return;
+            }
+
+            if (this.network.add(newService, newElement.get())) {
+                LOG.debug("Service already known: {}", newService);
+                return;
+            }
+
+            this.addService.accept(new ServiceEntry<>(newService, newElement.get())); // TODO fooker: this smells
+        }));
+    }
+
+    @EventHandler(uei = EventConstants.SUSPEND_POLLING_SERVICE_EVENT_UEI)
+    public void nodeRemovePollableServiceHandler(final IEvent event) {
+        final Service service = Service.fromEvent(event);
+
+        final Optional<ServiceEntry<E>> entry = this.network.remove(service);
+        if (!entry.isPresent()) { // TODO fooker: this smells
+            LOG.debug("Service not know: {}", service);
+            return;
+        }
+
+        entry.ifPresent(this.delService);
+
+    }
+
+    @EventHandler(ueis = {EventConstants.NODE_DELETED_EVENT_UEI,
+                          EventConstants.DUP_NODE_DELETED_EVENT_UEI})
+    public void nodeDeletedHandler(final IEvent event) {
+        final Node node = Node.fromEvent(event);
+
+//        this.getQueryManager().closeOutagesForNode(event.getTime(), event.getDbid(), node.nodeId);
```

Comment:
there are quite some lines of commented code in this class. Not sure if you want to remove right away or do you still need it for a later refactoring?